crates/miden-lib/src/transaction/inputs.rs

```diff
@@ -178,251 +27,408 @@ enum AccountInputsType {
-/// - The account storage commitment |-> storage slots and types vector.
-/// - The account code commitment |-> procedures vector.
-/// - The leaf hash |-> (key, value), for all leaves of the partial vault.
-/// - If present, the Merkle leaves associated with the account storage maps.
-/// - account_id |-> account_seed, when `account_inputs_type` is [`AccountInputsType::NativeNew`].
-/// - account_id |-> [ID_AND_NONCE, VAULT_ROOT, STORAGE_COMMITMENT, CODE_COMMITMENT] when
-///   `account_inputs_type` is [`AccountInputsType::Foreign`].
-fn add_account_to_advice_inputs(
-    account: &PartialAccount,
-    account_inputs_type: AccountInputsType,
-    advice_inputs: &mut AdviceInputs,
-) -> Result<(), TransactionInputError> {
-    let storage_header = account.storage().header();
-    let account_code = account.code();
-    let storage_proofs = account.storage().storage_map_proofs();
-    let account_id = account.id();
-
-    let account_header = AccountHeader::from(account);
-
-    // --- account storage ----------------------------------------------------
-
-    advice_inputs.extend_map([
-        // STORAGE_COMMITMENT -> [[STORAGE_SLOT_DATA]]
-        (storage_header.compute_commitment(), storage_header.as_elements()),
-    ]);
-
-    // load merkle nodes for storage maps
-    for proof in storage_proofs {
-        // Extend the merkle store and map with the storage maps
-        advice_inputs.extend_merkle_store(
-            proof
-                .path()
-                .inner_nodes(proof.leaf().index().value(), proof.leaf().hash())
-                .map_err(|err| {
-                    TransactionInputError::InvalidMerklePath(
-                        format!("foreign account ID {} storage proof", account_id).into(),
-                        err,
-                    )
-                })?,
-        );
-        // Populate advice map with Sparse Merkle Tree leaf nodes
-        advice_inputs
-            .extend_map(core::iter::once((proof.leaf().hash(), proof.leaf().to_elements())));
+// TRANSACTION ADVICE INPUTS
+// ================================================================================================
+
+/// Advice inputs wrapper for inputs that are meant to be used exclusively on the transaction
+/// kernel.
+#[derive(Default, Clone, Debug)]
+pub struct TransactionAdviceInputs(AdviceInputs);
+
+impl TransactionAdviceInputs {
+    /// Creates a [`TransactionAdviceInputs`].
+    ///
+    /// The created advice inputs will be populated with the data required for executing a
+    /// transaction with the specified inputs and arguments. This includes the initial account, an
+    /// optional account seed (required for new accounts), and the input note data, including
+    /// core note data + authentication paths all the way to the root of one of partial
+    /// blockchain peaks.
+    pub fn new(
+        tx_inputs: &TransactionInputs,
+        tx_args: &TransactionArgs,
+    ) -> Result<Self, TransactionInputError> {
```

Comment:
I was originally thinking that the constructor would look something like:
```rust
pub fn new(init_inputs: Option<AdviceInputs>) -> Self {
    ...
}
```
And then we'd have a separate method adding tx inputs/args. But I think this probably works just as well (if not better).